test/powershell/Language/Scripting/DeserializedMethods.Tests.ps1

```diff
@@ -0,0 +1,40 @@
+﻿Describe "DeserializedMethods" -Tags "CI" {
+
+    BeforeAll {
+        try {
+            $file = New-Item "TestDrive:/deserializedMethod$PID.txt" -Force            
+        } catch {
+            Remove-Item $file -force -ea silentlycontinue
+            return
+        }
+    }
+
+    AfterAll {
+	        Remove-Item $file -force -ea silentlycontinue
+        }    
+    It "Deserialized objects shouldn't ever have any methods (unless they are primitive known types)" {
+        $a = new-object collections.arraylist
+        $null = $a.Add(1)
+        $null = $a.Add(2)
+        $null = $a.Add(3)
+        
+	    # using linkedlist that implements IEnumerable,
+	    # but doesn't implement IList or IList<T>
+	    $x = new-object collections.generic.linkedlist[int]
+	    $null = $x.Add(123)
+	    $null = $x.Add(456)
+
+	    export-clixml -in $x -path $file
+	    $d = import-clixml $file	
+	    
+	    try
+	    {
+		    $null = $d.Add(789)
+            Throw "Execution OK"
+	    }
```

Comment:
We should be using better messages, it is confusing when the error logs say "ExecutionOK".

A better error message might be "exception expected, execution should not have reached here"